rules/loading.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2013 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rules
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/pkg/rulefmt"
+	"github.com/prometheus/prometheus/promql/parser"
+)
+
+// Grouploader is an interface for loading and parsing rules from arbitrary sources.
+type GroupLoader interface {
+	// Load resolves a list of parsed rules from a set of identifiers.
+	Load(identifiers ...string) ([]*ParsedGroup, []error)
+}
+
+// ParsedGroup binds a set of ParsedRules with an evaluation interval.
+// 0 indicates that the global default interval should be used.
+type ParsedGroup struct {
```

Comment:
We already have this information in the RuleGroup&Rule structs - there's no need to add an additional basically identical set of structs